### PR TITLE
Enable timeline page for Dart CLI apps.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Dart CLI apps will have access to the Full DevTools timeline. When connected to a Dart CLI app we remove the option to switch between timeline modes.

Screen Shot 2019-11-06 at 10 41 33 AM

### CODE REVIEW

File: packages/devtools_app/lib/src/main.dart [hunk 1]
```diff
@@ -233,7 +234,8 @@ class HtmlPerfToolFramework extends HtmlFramework {
           : notRunningFlutterMsg,
     ));
     addScreen(HtmlTimelineScreen(
-      enabled: isFlutterApp && !isFlutterWebApp,
+      isDartCliApp ? TimelineMode.full : TimelineMode.frameBased,
+      enabled: (isFlutterApp && !isFlutterWebApp) || isDartCliApp,
```
Comment (jacob314) [inline packages/devtools_app/lib/src/main.dart]: this enabled case can probably be simplified to `app.isRunningOnDartVM`

File: packages/devtools_app/lib/src/timeline/html_timeline_screen.dart [hunk 2]
```diff
@@ -430,10 +444,14 @@ class HtmlTimelineScreen extends HtmlScreen {
     // changes the value of [offlineMode], which the button states depend on.
     framework.exitOfflineMode();
 
-    // Revert to the previously selected mode on offline exit. We already
-    // cleared the timeline data - do not repeat this action.
+    // Revert to the previously selected mode on offline exit, unless there is
+    // only one allowed mode for the connected app. We already cleared the
+    // timeline data - do not repeat this action.
+    final mode = await serviceManager.connectedApp.isDartCliApp
```
Comment (jacob314) [inline packages/devtools_app/lib/src/timeline/html_timeline_screen.dart]: nit: why don't you instead store the initial default value already passed in for the screen and use that here?

File: packages/devtools_app/lib/src/main.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/main.dart]: done.

File: packages/devtools_app/lib/src/timeline/html_timeline_screen.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/html_timeline_screen.dart]: good idea. done.

**Comment by kenzieschmoll** [comment]
@DanTup Could you take a look at the latest [commit](https://github.com/flutter/devtools/pull/1309/commits/6def0bbe98172f5c0c1cd7c68ade7264fbc73932)? It fixes the previous integration test failures, but I want to make sure there isn't something I'm missing.

**Comment by DanTup** [comment]
> @DanTup Could you take a look at the latest commit? It fixes the previous integration test failures, but I want to make sure there isn't something I'm missing.

Yeah, that definitely looks like a bug - thanks!

**Pull request merged**